MoreLinq/Experimental/CurrentList.cs

```diff
@@ -0,0 +1,108 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2022 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+#if !NO_BUFFERS
+
+namespace MoreLinq.Experimental
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Represents a list that is the current view of a larger result and which
+    /// is updated in-place (thus current) as it is moved through the overall
+    /// result.
+    /// </summary>
+    /// <typeparam name="T">Type of elements in the list.</typeparam>
+
+    public interface ICurrentList<T> : IList<T> { }
```

Comment:
But it's not a read-only list so that would be misleading.

Reply:
Fair enough; I saw the `NotSupportedException`s and assumed it was. 

Might also add to doc to indicate that while the underlying elements may be modified as a scratch-space during the current iteration, any operations that affect the size are blocked in order to reuse the list in the next iteration?

Reply:
> Might also add to doc to indicate that while the underlying elements may be modified as a scratch-space during the current iteration, any operations that affect the size are blocked in order to reuse the list in the next iteration?

While that may be true for one implementation, I am not sure that's something that should be mandated at the abstraction level? If the abstraction was tied to `Batch` then that'd be fair, but I didn't see a reason to tie it unnecessarily. [It could potentially be used by other operators (like `Window` & co)](https://github.com/morelinq/MoreLINQ/issues/627). To be honest, this is the experimental part of the API. I am not even sure it should have _list_ in its name. It inherits from `IList<T>` so some operators can access it optimally (e.g. `ElementAt`), but that doesn't mean it should be modeled as a list.

Reply:
I've renamed it to `ICurrentBuffer<T>` with 0fc8901e53efa1cd11eac489d5bebcab4e5d0d0a. I think it's a little more accurate.  It doesn't evoke the idea of a list and buffer implies that current content is in memory. What do you think?

Reply:
I agree with `ICurrentBuffer<T>` - semantics are closer to intended use of the collection.